### DISCUSSION TITLE
Failed to save avatar to proxy

### DISCUSSION THREAD

**Discussion by Ivan-Chupin** [umputun/remark42#978] [Ideas]
If an account on a social network does not contain an avatar, then when registering through it, Remark42 reports an error `{"error": "failed to save avatar to proxy"}`. Not all users have a social media profile picture. The solution to this error is, for example, creating an avatar as for an anonymous user.

**Comment by umputun** [comment]
avatars handling, as well as anything else related to auth is a part of https://github.com/go-pkgz/auth not the remark42 itself. Could you pls open a ticket a provide steps to reproduce the issue?

**Reply by umputun** [reply]
btw, this library already trying to do exactly what you suggesting, i.e. making identicon avatars if user's picture not set in the profile.

<img width="958" alt="image" src="https://user-images.githubusercontent.com/535880/117195262-b523cd00-adaa-11eb-8006-1adff73f52db.png">

**Comment by Ivan-Chupin** [comment]
You can simulate the error by deleting the avatar in the social network account, having previously denied this access for Remark42 in this social network. Then try to re-register as a new user in Remark42 through the social network. I have tested Microsoft account.

P.S. I have a guess that this error is related to the error [Invalid avatar link when importing from Disqus](https://github.com/umputun/remark42/issues/987).

**Comment by umputun** [comment]
I'm having a hard time reproducing the issue. Can you pls run remark42 with `--dbg` (or env `DEBUG=true`) and show messages related to the login and avatar failure?

**Reply by Ivan-Chupin** [reply]
Because the data, in my opinion, contains confidential information, it is sent to your email.

**Reply by umputun** [reply]
thx, this was helpful and confirmed what I suspected. Will address it

**Comment by umputun** [comment]
@Ivan-Chupin This is the ticket I have opened https://github.com/go-pkgz/auth/issues/85

**Comment by umputun** [comment]
the current master should be fixed. pls give it a try

**Reply by Ivan-Chupin** [reply]
How can I try this if there is no release higher than 1.8.1? Or I don't understand something and release 1.8.1 is recompiled with a fixed error?

**Reply by umputun** [reply]
each commit to the master automatically makes a fresh docker image `umputun/remark42:master`, so if you use it with docker this image is ready for you.

to get the binary you can build it with `make release` (will take about 4-5mins) or just copy from the container, like this:

```
docker run --name=remark42.bin -d -v /tmp:/srv/var umputun/remark42:master /srv/remark42 server --secret=1234 --url=http://localhost
docker cp remark42.bin:/srv/remark42 .
docker rm -f remark42.bin
```

This will copy `remark42` binary (x86 version) to your current dir

```
./remark42 -h
remark42 master-19ba43b-20210507-23:39:58
Usage:
  remark42 [OPTIONS] <command>

Application Options:
      --url=    url to remark [$REMARK_URL]
      --secret= shared secret key used to sign JWT, should be a random, long, hard-to-guess string [$SECRET]
      --dbg     debug mode [$DEBUG]

Help Options:
  -h, --help    Show this help message

Available commands:
  avatar
  backup
  cleanup
  import
  remap
  restore
  server
```

**Reply by Ivan-Chupin** [reply]
I do not use Docker and am not familiar with Golang and do not want to produce unnecessary entities. From what you wrote, I did not understand anything, so I will wait for a new release.

**Reply by umputun** [reply]
attached binaries from the master in case you want to check it out:

[remark42.darwin-amd64.tar.gz](https://github.com/umputun/remark42/files/6446389/remark42.darwin-amd64.tar.gz)
[remark42.freebsd-amd64.tar.gz](https://github.com/umputun/remark42/files/6446390/remark42.freebsd-amd64.tar.gz)
[remark42.linux-386.tar.gz](https://github.com/umputun/remark42/files/6446391/remark42.linux-386.tar.gz)
[remark42.linux-amd64.tar.gz](https://github.com/umputun/remark42/files/6446392/remark42.linux-amd64.tar.gz)
[remark42.linux-arm64.tar.gz](https://github.com/umputun/remark42/files/6446393/remark42.linux-arm64.tar.gz)
[remark42.windows-amd64.zip](https://github.com/umputun/remark42/files/6446394/remark42.windows-amd64.zip)

**Reply by Ivan-Chupin** [reply]
The new version works fine. I tested it on a Microsoft account. If there is no avatar in the account, then Remark42 creates a random one, as for anonymous registrations.

**Reply by umputun** [reply]
thx for confirming!
